src/middlewared/middlewared/plugins/jbof/crud.py

```diff
@@ -0,0 +1,434 @@
+            except Exception:
+                pass
+            verrors.add('jbof_create.mgmt_ip1', f'Failed to add JBOF: {e}')
+        verrors.check()
+
+        data['id'] = await self.middleware.call(
+            'datastore.insert', self._config.datastore, data,
+            {'prefix': self._config.datastore_prefix})
+
+        return await self.get_instance(data['id'])
+
+    @accepts(
+        Int('id', required=True),
+        Patch(
+            'jbof_create', 'jbof_update',
+            ('attr', {'update': True})
+        )
+    )
+    async def do_update(self, id_, data):
+        """
+        Update JBOF of `id`
+        """
+        old = await self.get_instance(id_)
+        new = old.copy()
+        new.update(data)
+        verrors, data = await self.validate(new, 'jbof_update', old)
+        verrors.check()
+
+        if old['uuid'] != new['uuid']:
+            self.middleware.logger.debug('Changed UUID of JBOF')
+            await self.middleware.call('jbof.unwire_dataplane', old['mgmt_ip1'], old['index'])
+            await self.middleware.call('jbof.hardwire_dataplane', new['mgmt_ip1'], new['index'], 'jbof_update.mgmt_ip1', verrors)
+
+        await self.middleware.call(
+            'datastore.update', self._config.datastore, id_, new,
+            {'prefix': self._config.datastore_prefix})
+
+        return await self.get_instance(id_)
+
+    @accepts(Int('id'))
+    async def do_delete(self, id_):
+        """
+        Delete a JBOF by ID.
+        """
+        # Will make a best-effort un tear down existing connections / wiring
+        # To do that we first need to fetch the config.
+        data = await self.get_instance(id_)
+        await self.middleware.run_in_thread(self.ensure_redfish_client_cached,
+                                            data['mgmt_ip1'],
+                                            data.get('mgmt_username'),
+                                            data.get('mgmt_password'))
+        try:
+            await self.middleware.call('jbof.unwire_dataplane', data['mgmt_ip1'], data['index'])
+        except Exception:
+            self.middleware.logger.debug('Unable to unwire JBOF @%r', data['mgmt_ip1'])
+
+        # Now delete the entry
+        response = await self.middleware.call('datastore.delete', self._config.datastore, id_)
+        # await self.middleware.call('service.restart', 'ntpd')
+        return response
+
+    @private
+    def ensure_redfish_client_cached(self, mgmt_ip, username=None, password=None):
+        """Synchronous function to ensure we have a redfish client in cache."""
+        try:
+            RedfishClient.cache_get(self.middleware, mgmt_ip)
+        except KeyError:
+            # This could take a while to login, etc ... hence synchronous wrapper.
+            redfish = RedfishClient(f'https://{mgmt_ip}', username, password)
+            RedfishClient.cache_set(mgmt_ip, redfish)
+
+    @private
+    async def licensed(self):
```

Comment:
Let's include this in either `system.info` output or some other appropriate public API endpoint for webui consumption.